frontend/src/app/TelemetryService.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  ApplicationInsights,
+  SeverityLevel,
+} from "@microsoft/applicationinsights-web";
+import { ReactPlugin } from "@microsoft/applicationinsights-react-js";
+import { useHistory } from "react-router";
+
+let reactPlugin: ReactPlugin | null = null;
+let appInsights: ApplicationInsights | null = null;
+
+const createTelemetryService = () => {
+  const initialize = (browserHistory?: ReturnType<typeof useHistory>) => {
+    const instrumentationKey = process.env.REACT_APP_APPINSIGHTS_KEY;
+
+    if (!instrumentationKey) {
+      console.warn("Instrumentation key not provided");
+      return;
+    }
+
+    reactPlugin = new ReactPlugin();
+
+    appInsights = new ApplicationInsights({
+      config: {
+        instrumentationKey,
+        extensions: [reactPlugin],
+        loggingLevelConsole: 2,
```

Comment:
**thought:** Related to the above conversation about infinite loops: another option for that could be passing `loggingLevelConsole: 0` here to the `ApplicationInsights` config at initialization? I'm a little fuzzy abt the AppInsights logic here, but I know that at least will cover `throwInternal`, if not explicitly `warnToConsole`

Reply:
🎉